lib/child_process.js

```diff
@@ -536,6 +538,31 @@ function copyProcessEnvToEnv(env, name, optionEnv) {
   }
 }
 
+let permissionModelFlagsToCopy;
+
+function getPermissionModelFlagsToCopy() {
+  if (permissionModelFlagsToCopy === undefined) {
+    permissionModelFlagsToCopy = [...permission.availableFlags(), '--permission'];
+  }
+  return permissionModelFlagsToCopy;
+}
+
+function copyPermissionModelFlagsToEnv(env, key, args) {
+  // Do not override if permission was already passed to file
+  if (args.includes('--permission') || (env[key] && env[key].indexOf('--permission') !== -1)) {
```

Comment:
You can simplify this expression a bit with...

```js
env?.[key]?.indexOf(...)
```

Reply:
How? We need to compare the result with `!== -1` and also handle the case when the key is `undefined`. Leaving it as `env?.[key]?.indexOf('--permission')` might return `0`, which is truthy in this case, so I don't see how using `env?.[key]?.indexOf(...)` would simplify this.

Reply:
I don't think you are correct

```
!~({'foo': '--permission'})?.['foo'].indexOf('--permission')
false
!~({'foo': '--permission'})?.['foo2']?.indexOf('--permission')
false
({'foo': '--permission'})?.['foo'].indexOf('--permission')
0
```